assets/src/application/ui/layout/entityList/pagination/useEntityPagination.js

```diff
@@ -1,35 +1,36 @@
-/**
- * External imports
- */
-import { useEffect, useState } from '@wordpress/element';
-
+import { useState } from 'react';
+import { slice } from 'ramda';
 const EMPTY_ARRAY = [];
 
-/**
- * tracks current page for any paginated list
- *
- * @param {number} perPage
- * @param {BaseEntity[]} entities
- * @return {Object} - currentPage
- *                  - setCurrentPage - callback for changing currentPage
- *                  - paginatedEntities - array of entities for current page
- */
-const useEntityPagination = (perPage, entities) => {
-	const [currentPage, setCurrentPage] = useState(1);
-	const pageNumber = parseInt(currentPage, 10);
-	const lastPage = Math.ceil(entities.length / perPage);
-	const actualPageNumber = pageNumber <= lastPage ? pageNumber : lastPage;
-	useEffect(() => {
-		if (pageNumber > 1 && pageNumber > lastPage) {
-			setCurrentPage(lastPage);
-		}
-	}, [perPage, currentPage, entities.length]);
+const useEntityPagination = ({ entities, perPage }) => {
+	const initialPaginatedEntities = slice(0, perPage)(entities);
+	const [paginatedEntities, setCurrentPage] = useState(initialPaginatedEntities);
+	const total = entities.length;
+	// const pageNumber = parseInt(currentPage, 10);
+	// const lastPage = Math.ceil(entities.length / perPage);
+	// const actualPageNumber = pageNumber <= lastPage ? pageNumber : lastPage;
+
+	// useEffect(() => {
+	// 	if (pageNumber > 1 && pageNumber > lastPage) {
+	// 		setCurrentPage(lastPage);
+	// 	}
+	// }, [perPage, currentPage, entities.length]);
```

Comment:
Left these comments, because I was thinking that I might reuse smth on next iteration, if not, will remove update and clean up in next PR.